### ISSUE TITLE
Combat scripts should allow manual entry of rolls to reuse misconfigured rolls

### ISSUE THREAD

**Issue by phylll** [phylll/mychs-macro-magic#61]
When an attack or defense roll is mistakenly configured and needs to be repeated, the same actual rolls (the dice results) should be used.

Special case: the GM sometimes simply wants to redo *exactly* the same roll, kind of the reverse use case, to e.g. get rid of an ill-timed critical result. A "reroll" button right next to the approval button could be the UI for the GM to repeat exactly the same roll with all modifiers, the same attacker and target and weapon with a single click.

**Comment by phylll** [comment]
Here's the combined (but trimmed for redundancies) customize block for a regular attack the rolls of which I might want to reuse:

```
!mmm customize
!mmm   set cOwnID="-Mvd7ekEU8e_h0LcB-BY"
!mmm   set cTargetID="@{target|Angriffsziel|token_id}"
!mmm   set cSemiManualModifiers=?{Standard-Angriffsmodifikatoren|Keine +/-0,?{Keine +/-0 +/-...&vert;Echt keine +/-0&comma;1&vert;Von oben 70cm+ +2&comma;2&vert;Ziel liegt/ich stehe +4&comma;3&vert;Ziel wehrlos/überrascht +4&comma;5&vert;Ziel flieht panisch +4&comma;7&rbrace;|Spontan -4,?{Spontan -4 +/-...&vert;Nix weiter +/-0&comma;11&vert;Von oben 70cm+ +2&comma;22&vert;Ziel liegt/ich stehe +4&comma;33&vert;Ziel wehrlos/überrascht +4&comma;55&vert;Ziel flieht panisch +4&comma;77&rbrace;|Überstürzt -6,?{Überstürzt -6 +/-...&vert;Nix weiter +/-0&comma;13&vert;Von oben 70cm+ +2&comma;26&vert;Ziel liegt/ich stehe +4&comma;39&vert;Ziel wehrlos/überrascht +4&comma;65&vert;Ziel flieht panisch +4&comma;91&rbrace;|Völlig dunkel/geblendet -6,?{Völlig dunkel/geblendet -6 +/-...&vert;Nix weiter +/-0&comma;17&vert;Von oben 70cm+ +2&comma;34&vert;Ziel liegt/ich stehe +4&comma;51&vert;Ziel wehrlos/überrascht +4&comma;85&vert;Ziel flieht panisch +4&comma;119&vert;Spontan -4&comma;?{Keine +/-0 +/-...&amp;vert;Echt keine +/-0&amp;comma;187&amp;vert;Von oben 70cm+ +2&amp;comma;374&amp;vert;Ziel liegt/ich stehe +4&amp;comma;561&amp;vert;Ziel wehrlos/überrascht +4&amp;comma;935&amp;vert;Ziel flieht panisch +4&amp;comma;1309&amp;rbrace;&vert;Überstürzt -6&comma;?{Keine +/-0 +/-...&amp;vert;Echt keine +/-0&amp;comma;221&amp;vert;Von oben 70cm+ +2&amp;comma;442&amp;vert;Ziel liegt/ich stehe +4&amp;comma;663&amp;vert;Ziel wehrlos/überrascht +4&amp;comma;1105&amp;vert;Ziel flieht panisch +4&amp;comma;1547&amp;rbrace;&rbrace; }
!mmm   set cManualModifiers=?{Weitere spezielle Angriffsmodifikatoren|0}
&{template} [[1d20]] {{}}
!mmm   set cAttackRoll=$[[0]]
!mmm   set cGMSilentMode=false
!mmm end customize
!mmm customize
!mmm   set cVerbose = true
!mmm   set cWeaponLabel = "Feuermal (Langschwert)"
!mmm   set cWeaponName = "Feuermal"
!mmm   set cWeaponType = "Einhandschwert"
!mmm   set cMagicExtraDamageMarker = "status_all_for_one"
!mmm   set cMagicExtraDamage = "1D6"
!mmm   set cMagicExtraDamageLabel = "✨🔥"
!mmm end customize
%{MacroSheet|meleeAttack}
```

Now, the only thing we need is to repeat the same through a button, while setting `cAttackRoll` statically to the previously rolled number (or `roll(number)`) and add similar parameters for the other rolls (primarily the damage roll but also the second sharpshooting roll etc.). Should be easy to implement.

For the reverse use case (keep everything *but* the rolls), the same applies: we need to regenerate the config block with the modifiers and target params set statically, and either redo (3D rolls for players) or leave the rolls to the main script. Piece of 🍰!

**Comment by phylll** [comment]
Reusing previous rolls is implemented for `meleeAttack` in 85732b38aa683eeae3d69fe894294fee148d0bc6

**Comment by phylll** [comment]
Rerolling with constant modifiers, for the GM, implemented in eff8cb9f143df966f77b5ed4451829f2b08a829e
